src/braket/ir/jaqcd/instructions.py

```diff
@@ -899,3 +900,43 @@ class Type(str, Enum):
         kraus = "kraus"
 
     type = Type.kraus
+
+
+class StartPreserveBlock(CompilerDirective):
+    """
+    StartPreserveBLock is a compiler instruction to start the code block which
+    will preserve the instruction within StartPreserveBlock and EndPreserveBlock
+    from being modified in any way by the compiler.
+
+    Attributes:
+        type (str): The instruction type. default = "start_preserve_block". (type) is optional.
+            This should be unique among all instruction types.
+
+    Examples:
+        >>> StartPreserveBlock(directive=["StartPreserveBlock"])
+    """
+
+    class Type(str, Enum):
+        start_preserve_block = "start_preserve_block"
+
+    type = Type.start_preserve_block
+
+
+class EndPreserveBlock(CompilerDirective):
+    """
+    EndPreserveBLock is a compiler instruction to mark the end of the code block
+    which will preserve the instruction within StartPreserveBlock and EndPreserveBlock
+    from being  modified in any way by the compiler.
+
+    Attributes:
+        type (str): The instruction type. default = "end_preserve_block". (type) is optional.
+            This should be unique among all instruction types.
+
+    Examples:
+        >>> EndPreserveBlock(directive=["EndPreserveBlock"])
```

Comment:
This seems a little redundant. The information carried by the directive is the same as the class name and the type. Do we expect the directive to diverge from the instruction type ?

Reply:
Introduced the default directive for StartVerbatimBlock and EndVerbatimBlock classes to reduce the redundancy.